vendor.py

```diff
@@ -44,6 +73,7 @@
     libraries = cache['libraries']
     windows_options = cache['windows_options']
     cflags = cache['cflags']
+    optional_extensions = cache['optional_extensions']
 elif 'DUCKDB_NODE_BINDIR' in os.environ:
```

Comment:
No idea around these ifs and elses. Makes it very hard to reason about code with re-assigning variables. Haven't found explicit usage of it, but wasn't confident of deleting these.